### PR TITLE
Add project creation to DataSource

### PR DESCRIPTION
**Pull request by willsb**
Ref #798 

Squash whenever

### CODE REVIEW

**Comment by amulware** [comment]
This is not handling several properties of the project model. Why is that?

**Comment by willsb** [comment]
This is handling the properties that appear in the create view. More will be added as (and if) needed

**Comment by amulware** [comment]
We need at least the billable field, and maybe also some related to estimataes (see issue).

**Comment by amulware** [comment]
Only the billable field is needed and should be taken from the selected workspace default field.

**Comment by amulware** [comment]
Or be left at `null` for workspaces missingthe Pro feature.

**Comment by willsb** [comment]
So only pro workspaces have the `ProjectsBillableByDefault` flag?

**Comment by amulware** [comment]
Any workspace 'has' the field, but it should only be considered if they are Pro, yes.

File: Toggl.Foundation/Models/Project.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+            public DateTimeOffset? At { get; private set; }
+
+            public DateTimeOffset? ServerDeletedAt { get; private set; }
+
+            public bool IsDeleted { get; private set; }
+
+            public long? ClientId { get; private set; }
+
+            public long? TemplateId { get; private set; }
```
Comment (amulware) [inline Toggl.Foundation/Models/Project.cs]: Let's remove this until we need it.

File: Toggl.Foundation/DTOs/CreateProjectDTO.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    {
+        public string Name { get; set; }
+
+        public string Color { get; set; }
+
+        public bool IsPrivate { get; set; }
+
+        public long? ClientId { get; set; }
+
+        public long? TemplateId { get; set; }
```
Comment (amulware) [inline Toggl.Foundation/DTOs/CreateProjectDTO.cs]: :fire: until needed

**Pull request merged**